### PR TITLE
add msvcp140.dll to the wheel on windows

### LINKED ISSUE #13293: Include compiled extension binaries into windows wheel
At the moment, when installing Ray on Windows, despite there being a wheel, it requires compiler to be installed. This is well documented in the [Windows Dependencies](https://docs.ray.io/en/master/installation.html#windows-dependencies) section.

One of the main benefits of the wheel format however, is to distribute pre-built binaries - so that users would be able to install packages without compiler present. We could as well just install from source if we had compiler setup.. but the problem is that in certain Corporate environments it's just not feasible for most Windows users to have the compiler installed.

My question is - is it possible to compile the necessary extension(s) as part of the build and include them in the wheel - so that Windows users without C++ compiler could use ray (with `pip install`)?

Thanks!

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Closes #13293. As described in the issue, on a minimal windows image the support dll `msvcp140.dll` is needed to load `_raylet.pyd`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -558,6 +558,11 @@ def pip_run(build_ext):
     copied_files = 0
     for filename in setup_spec.files_to_include:
         copied_files += copy_file(build_ext.build_lib, filename, ROOT_DIR)
+    if sys.platform == 'win32':
+        print('copy_file uses', build_ext.build_lib, ROOT_DIR)
+        shutil.copy(r'c:\Windows\system32\msvcp140.dll',
+                        os.path.join(build_ext.build_lib, 'ray'))
```
Comment (richardliaw) [inline python/setup.py:564]:
```suggestion
        print("copy_file uses", build_ext.build_lib, ROOT_DIR)
        shutil.copy(r"c:\Windows\system32\msvcp140.dll",
                        os.path.join(build_ext.build_lib, "ray"))
```

Comment (richardliaw) [reply python/setup.py:564]: We use double quotes for literals

Comment (mattip) [reply python/setup.py]: thanks, fixed and removed the debug print

**Comment by mattip** [comment]
On the one hand this is a very small change, on the other it is a bit obscure how the dll ends up in the wheel. I guess `git grep msvcp140` will find it, is that discoverable enough or should I make a `copy_system_dlls()` function to emphasize it?

**Comment by richardliaw** [comment]
Explicit emphasis and comments will help maintainers moving forward!

**Review by pcmoritz — APPROVED** [review]
Thanks for doing this, this is great! Looks like the windows tests failed for unrelated reasons, re-running them.

**Comment by pcmoritz** [comment]
@mattip Looks like there is an unrelated doc building failure, can you merge master into this PR? that should make it go away!

**Comment by mattip** [comment]
rebased off master

**Review by richardliaw — APPROVED** [review]

**Pull request merged**